src/test/java/org/folio/rest/api/InstanceStorageTest.java

```diff
@@ -1783,6 +1805,510 @@ public void cannotPostSynchronousBatchWithDuplicateId() throws Exception {
+    instanceToCreate.remove("hrid");
+
+    setInstanceSequence(1);
+
+    createInstance(instanceToCreate);
+
+    final Response createdInstance = getById(id);
+
+    assertThat(createdInstance.getJson().getString("hrid"), is("in00000001"));
+
+    final JsonObject instanceToCreateWithSameHRID = nod(UUID.randomUUID());
+    instanceToCreateWithSameHRID.put("hrid", "in00000001");
+
+    final CompletableFuture<Response> createCompleted = new CompletableFuture<>();
+
+    client.post(instancesStorageUrl(""), instanceToCreateWithSameHRID, TENANT_ID,
+      text(createCompleted));
+
+    final Response response = createCompleted.get(5, SECONDS);
+
+    assertThat(response.getStatusCode(), is(400));
+    assertThat(response.getBody(),
+        is("duplicate key value violates unique constraint \"instance_hrid_idx_unique\": Key (lower(f_unaccent(jsonb ->> 'hrid'::text)))=(in00000001) already exists."));
+
+    log.info("Finished cannotCreateInstanceWithDuplicateHRID");
+  }
+
+  @Test
+  public void cannotCreateInstanceWithHRIDFailure() throws Exception {
+    log.info("Starting cannotCreateInstanceWithHRIDFailure");
+
+    final UUID id = UUID.randomUUID();
+    final JsonObject instanceToCreate = smallAngryPlanet(id);
+    instanceToCreate.remove("hrid");
+
+    setInstanceSequence(99999999);
+
+    createInstance(instanceToCreate);
+
+    final Response createdInstance = getById(id);
+
+    assertThat(createdInstance.getJson().getString("hrid"), is("in99999999"));
+
+    final JsonObject instanceToFail = nod(UUID.randomUUID());
+    instanceToFail.remove("hrid");
+
+    final CompletableFuture<Response> createCompleted = new CompletableFuture<>();
+
+    client.post(instancesStorageUrl(""), instanceToFail, TENANT_ID, text(createCompleted));
+
+    final Response response = createCompleted.get(5, SECONDS);
+
+    assertThat(response.getStatusCode(), is(500));
+    assertThat(response.getBody(),
+        is("ErrorMessage(fields=[(Severity, ERROR), (V, ERROR), (SQLSTATE, 2200H), (Message, nextval: reached maximum value of sequence \"hrid_instances_seq\" (99999999)), (File, sequence.c), (Line, 700), (Routine, nextval_internal)])"));
+
+    log.info("Finished cannotCreateInstanceWithHRIDFailure");
+  }
+
+  @Test
+  public void cannotChageHRIDAfterCreation() throws Exception {
+    log.info("Starting cannotChageHRIDAfterCreation");
+
+    final UUID id = UUID.randomUUID();
+    final JsonObject instanceToCreate = smallAngryPlanet(id);
+    instanceToCreate.remove("hrid");
+
+    setInstanceSequence(1);
+
+    createInstance(instanceToCreate);
+
+    final JsonObject instance = getById(id).getJson();
+    final String expectedHRID = "in00000001";
+
+    assertThat(instance.getString("hrid"), is(expectedHRID));
+
+    instance.put("hrid", "testHRID");
+
+    final JsonObject updatedInstance = updateInstance(instance).getJson();
+
+    assertThat(updatedInstance.getString("hrid"), is(expectedHRID));
```

Comment:
Does this mean that any new HRID is silently ignored, rather than the whole request being rejected?

Reply:
Yes. We could do this either way. Does it make more sense to reject the request with a message stating the `hrid` field cannot be changed or ignore the change and complete the update? I suppose if the only change is the `hrid` field, this could be a bit confusing.

Reply:
If the `hrid` was readonly in the schema because we used different schema for reads and writes, then I'd suggest ignoring it would be best.

As it isn't, I think if it is no more work to implement, then refusing the request is better. It allows the client to understand. Most of the time, the client should be basing their PUT request on a recently fetched representation, so it should be the same.